app/src/main/java/org/listenbrainz/android/viewmodel/ListensViewModel.kt

```diff
@@ -141,7 +147,10 @@ class ListensViewModel @Inject constructor(
     private fun updateTrackCoverArt(playerState: PlayerState) {
         // Get image from track
         assertAppRemoteConnected()?.imagesApi?.getImage(playerState.track.imageUri, com.spotify.protocol.types.Image.Dimension.LARGE)?.setResultCallback { bitmapHere ->
-            bitmap = bitmapHere
+            bitmap =ListenBitmap(
+                bitmap=bitmapHere,
+                id = playerState.track.uri
```

Comment:
I think we should have just made a separate variable here. Why store it with the bitmap like this?

Reply:
Actually, bitmap is updated from event callback, which is called every second to get song position and to avoid updating of bitmap again and again, the function will only run when uri which it store is different from the current song.